docs/book/en/src/dev/rust/wasinn.md

```diff
@@ -0,0 +1,158 @@
+# Neural Network for WASI
+
+In WasmEdge, we implemented the [WASI-NN](https://github.com/WebAssembly/wasi-nn) (Neural Network for WASI) proposal to allows WebAssembly programs to access machine learning(ML) functions, in the fashion of "graph" loader API. [WASI-NN](https://github.com/WebAssembly/wasi-nn) allows users to "load, bind, and evaluate" models by functions:
+
+* [`load`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L108-L118) a model using variable opaque byte arrays
+* [`init_execution_context`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L125-L129) and bind some tensors to it using [`set_input`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L134-L142)
+* [`compute`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L165) the ML inference using the bound context
+* retrieve the inference result tensors using [`get_output`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L147-L160)
+
+You can find more detail about the WASI-NN proposal in [Reference](#Reference).
+
+In this section, we will use [an Rust example project](https://github.com/gusye1234/Wasi-NN-Examples) ![badge](https://github.com/gusye1234/Wasi-NN-Examples/actions/workflows/main.yaml/badge.svg) to demonstrate how to use the WASI-NN api and run a image classification demo.
+
+## Prerequisite
+
+Currently, WasmEdge used OpenVINO™ as the backend implementation. For this demo, you need to install [OpenVINO™](https://docs.openvino.ai/2021.4/openvino_docs_install_guides_installing_openvino_linux.html#)(2021) and [Rust](https://www.rust-lang.org/tools/install).
+
+By default, we don't enable any WASI-NN backend in WasmEdge. To enable the OpenVINO™ backend, we need to [building the WasmEdge from source](../../extend/build.md) with one more compiler flag `-DWASMEDGE_WASINN_BUILD_OPENVINO=ON` to enable the OpenVINO™ backend:
+
+```sh
+...
+cmake -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_WASINN_BUILD_OPENVINO=ON .. && make -j
+```
+(*Make sure you have configured the OpenVINO™ environment correctly with `source /opt/intel/openvino_2021/bin/setupvars.sh `* )
+
+For Rust, make sure to add `wasm32-wasi` target with
+
+```shell
+rustup target add wasm32-wasi
+```
+
+## Run
```

Comment:
It's better to setup the examples and run with commands.
Please list the commands for both `cargo build`, `wasmedge`, etc.

Reply:
I'm a little bit confused about this.
Do you mean instead of using a bash file to build and run, we should expose the building and running process in this section?

Reply:
Sure. You can reserve the `build.sh` to quickly execution, but we should explain the entire steps.

Reply:
Got it. Thanks